src/lib/php/Dao/LicenseDao.php

```diff
@@ -713,4 +713,38 @@ public function getLicenseObligations($licenseLists, $candidate = false)
       return $ObligationRef;
     }
   }
+
+  /**
+   * @param string $licenseShortname
+   * @param int|null $groupId
+   * @param int|null $userId
+   * @return null
+   */
+  public function deleteLicenseByShortName($licenseShortname, $groupId=null, $userId=null)
+  {
+    $extraCondition = "";
+    $condition = 'rf_shortname=$1';
+    $param = array($licenseShortname);
+    $row = $this->dbManager->getSingleRow(
+      "DELETE FROM ONLY license_ref WHERE $condition RETURNING rf_shortname",
+      $param, __METHOD__ . ".$condition.only");
+
+    if (false === $row && isset($groupId)) {
+      if (!empty($userId)) {
+        $param[] = $userId;
+        $extraCondition = "AND group_fk IN (SELECT group_fk FROM group_user_member WHERE user_fk=$".count($param).")";
+      }
+      if (is_int($groupId) && empty($userId)) {
+        $param[] = $groupId;
+        $extraCondition = "AND group_fk=$".count($param);
+      }
+      $row = $this->dbManager->getSingleRow(
+        "DELETE FROM license_candidate WHERE $condition $extraCondition RETURNING rf_shortname",
+        $param, __METHOD__ . ".$condition.group");
+    }
+    if (false === $row) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
There is no return statement if logic did not enter `if (false === $row && isset($groupId))` condition. It will make function return `bool|null`.